core/src/transfer/moonlight.rs

```diff
@@ -274,6 +302,25 @@ impl Transaction {
         Some(stripped_transaction)
     }
 
+    /// Creates a modified clone of this transaction if it contains a Blob,
+    /// clones all fields except for the Blob, where its hash is set as Memo.
+    ///
+    /// Returns none if the transaction is not a Blob transaction.
+    #[must_use]
+    pub fn convert_blob(&self) -> Option<Self> {
```

Comment:
I have only nit remark to the name of this function: blob_to_memo would be better in my opinion, or sth else but revealing more what it does (e.g.: strip_blob_to_memo)

Reply:
I would name it `strip_off_blob` to be consistent with the similar `strip_off_bytecode` function

Reply:
Additionally I would implement the `blob_to_memo` functionality on the `TransactionData` directly to avoid code duplication

Reply:
> Maybe the `blob_to_memo` functionality can be implemented directly on the `TransactionData` to avoid code duplication

It's ok, we can have it too
We only need to propagate the mutable accessors one level up.

If you agree, I would like to test your suggested approach in a separate PR